sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3403,9 +3419,16 @@ pi_result piMemRelease(pi_mem Mem) {
       auto Buf = static_cast<_pi_buffer *>(Mem);
       if (!Buf->isSubBuffer()) {
         if (enableBufferPooling()) {
-          PI_CALL(piextUSMFree(Mem->Context, Mem->getZeHandle()));
+          pi_platform Plt = Mem->Context->Devices[0]->Platform;
+          std::unique_lock<std::mutex> ContextsLock(Plt->ContextsMutex,
+                                                    std::defer_lock);
+          if (IndirectAccessTrackingEnabled)
+            ContextsLock.lock();
+          return USMFreeHelper(Mem->Context, Mem->getZeHandle(),
+                               Mem->OwnZeMemHandle);
```

Comment:
maybe just pass "pi_mem Mem" instead?

Reply:
This function is used for pi_buffers as well as for regular USM allocations which don't have corresponding pi_mem object.